### PR TITLE
Cache Interfaces

### LINKED ISSUE #8: cache types
wondering whether cache types will be added here in future?

### PR DESCRIPTION
**Pull request by ashnewmanjones**
Interfaces for the caches API. Based the implementation off of issue #8 adding the delete method and options argument.

closes #8

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -153,6 +153,34 @@ interface Request {
...
+  default: {
+      /**
+       * Adds to the cache a response keyed to the given request.
+       * Returns a promise that resolves to undefined once the cache stores the response.
+       */
+      put(request: Request | string, response: Response): Promise<undefined>;
+      /**
+       * Returns a promise wrapping the response object keyed to that request.
+       */
+      match(request: Request | string, options: CacheOptions): Promise<Response | undefined>;
```
Comment (jrf0110) [inline index.d.ts]: I thiiiink that `options` should be optional (`options?: CacheOptions`)

File: index.d.ts [hunk 2]
```diff
@@ -153,6 +153,34 @@ interface Request {
   }
 }
 
+interface CacheOptions {
+  /**
+   * Consider the request method a GET regardless of its actual value.
+   */
+  ignoreMethod: boolean;
```
Comment (jrf0110) [inline index.d.ts]: This should probably be optional as well

**Comment by ashnewmanjones** [comment]
Oops my mistake, made the value optional.

**Comment by jrf0110** [comment]
Wow @ashnewmanjones I'm super sorry we never got this merged. Can you rebase master, fix the conflicts, and push back to your branch?

**Comment by ispivey** [comment]
@ashnewmanjones I'm closing this in favor of https://github.com/cloudflare/workers-types/pull/21 so we can fix this issue in a release ASAP, but thank you for pointing us in the right direction!

**Pull request closed without merging**